### PR TITLE
add crud for users table

### PR DESCRIPTION
**Pull request by florian-sabonchi**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,25 +1,21 @@
-import routes
+from typing import Union
 
 from fastapi import FastAPI
 
-app = FastAPI(
```
Comment (tanneberger) [inline app/main.py:6]: this is wrong dont remove it

File: app/schema/emergency.py [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
-class Status(enum.Enum):
+class Status:
     in_progress = 0
     cancelled = 1
     completed = 2
 
 
-class Emergency(Base):
-    __tablename__ = "emergency"
+class Emergencie(Model):
```
Comment (tanneberger) [inline app/schema/emergency.py]: class and table name are wrong fix it

File: app/schema/emergency.py [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
+    __tablename__ = "emergencies"
 
-    id: UUID = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
-    status: Status = Column(Enum(Status), default=Status.in_progress)
```
Comment (tanneberger) [inline app/schema/emergency.py:21]: dont remove type hints

File: app/schema/registered_user.py [hunk 4]
```diff
@@ -12,13 +14,39 @@ class Role(enum.Enum):
     user = 2
 
 
-class RegisteredUser(Base):
+class RegisteredUser(Model):
     __tablename__ = "registered_users"
 
-    id: UUID = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
```
Comment (tanneberger) [inline app/schema/registered_user.py:18]: dont remove type hints or make it proper with column type

File: app/schema/user.py [hunk 5]
```diff
@@ -1,11 +1,12 @@
-from datetime import datetime
```
Comment (tanneberger) [inline app/schema/user.py]: dont

File: app/schema/user.py [hunk 6]
```diff
@@ -14,8 +15,22 @@ class Role(enum.Enum):
...
+    id = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
+    created_at = Column(DateTime, default=datetime.datetime.utcnow)
+
+    @staticmethod
+    def get(user_id: uuid.UUID):
+        return session.query(User).get(user_id)
+
+    @staticmethod
+    def create():
+        session.add(User(id=uuid.uuid4(), created_at=datetime.datetime.utcnow()))
```
Comment (tanneberger) [inline app/schema/user.py]: datetime.utcnow()

File: app/schema/emergency.py [hunk 7]
```diff
@@ -1,25 +1,24 @@
-from datetime import datetime
```
Comment (tanneberger) [inline app/schema/emergency.py:1]: dont

**Review by tanneberger — CHANGES_REQUESTED** [review]

File: app/database.py [hunk 8]
```diff
@@ -3,13 +3,20 @@
...
 
 user = os.getenv("POSTGRES_USER")
 password = os.getenv("POSTGRES_PASSWORD")
 database = os.getenv("POSTGRES_DB")
 host = os.getenv("POSTGRES_HOST")
 port = os.getenv("POSTGRES_PORT", "5432")
 
-engine = create_engine(f"postgresql://{user}:{password}@{host}:{port}/{database}", echo=True)
+engine = create_engine(
+    f"postgresql://" f"{user}:{password}@{host}:{port}" f"/{database}",
```
Comment (felbinger) [inline app/database.py]: Why do you prefer this syntax (`f"A" f"B" f"C"`) instead of `f"ABC"`?

File: app/schema/emergency.py [hunk 9]
```diff
@@ -1,25 +1,59 @@
...
+from database import session
+
 from sqlalchemy import Column, Enum, DateTime, Float
 from sqlalchemy.dialects.postgresql import UUID
 
-from db import Base
+from database import Model
 
 
-class Status(enum.Enum):
```
Comment (felbinger) [inline app/schema/emergency.py:11]: isn't this an enumeration?

File: app/schema/user.py [hunk 10]
```diff
@@ -14,8 +15,22 @@ class Role(enum.Enum):
     user = 2
 
 
-class User(Base):
+class User(Model):
     __tablename__ = "users"
 
-    id: UUID = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
```
Comment (felbinger) [inline app/schema/user.py:20]: Should the typing be removed here? Tassilo ask to keep it above in the schema to register a user, why not here too?

**Review by felbinger — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 11]
```diff
@@ -28,9 +36,9 @@
 app.include_router(routes.user.user_router)
 app.include_router(routes.emergency.emergency_router)
 
+Model.metadata.create_all(bind=engine)
```
Comment (felixevers) [inline app/main.py]: move to fastapi preflight startup function

**Review by felixevers — CHANGES_REQUESTED** [review]
Well done! Just this one thing right there!

**Pull request merged**